### DISCUSSION TITLE
Monitor devices in your ISM neighborhood with rtl_watch

### DISCUSSION THREAD

**Discussion by hdtodd** [merbanan/rtl_433#3021] [Show and tell]
rtl_watch reports the signal characteristics of the ISM devices in your neighborhood: SNR, signal frequency, inter-transmission gap times, and packets per transmission.  

Earlier versions reported incorrect statistics as packets may have been dropped during processing.

v3.0 corrects that problem by using a message queue to separate the packet collection thread from the packet processing thread.  It's available at http://github.com/hdtodd/rtl_watch

**Comment by zuckschwerdt** [comment]
Very nice. You could also explore the HTTP API (see https://github.com/merbanan/rtl_433/blob/master/examples/rtl_433_http_stream.py and https://github.com/merbanan/rtl_433/blob/master/examples/rtl_433_http_events.py). With that you can optionally set `raw_mode` to receive undecoded events also, for perhaps a more detailed breakdown of general traffic vs decoded events.
Let me know where the API can be improved.

**Reply by hdtodd** [reply]
I saw your suggestion last year but couldn't explore further at the time.

As I understand it, using the '-F http' option allows the operator of the system to avoid installing and managing mosquitto and instead runs a local http service from rtl_433.  

Is that right? 

If so, then there would definitely be less complexity for the operator of the system.  I'll need to explore with some small test examples to understand how complicated it would be to introduce that option into my codes, but if I understand it correctly, it's worth a look.

David

**Reply by zuckschwerdt** [reply]
Yes, you need `-F http`, but that output can run along all other outputs. It may someday even become a default.

You might need to handle disconnects with a reconnect (which Paho does transparently), but you can just connect to the HTTP endpoint and read events.

You might not even need a HTTP lib, just opening the TCP port (e.g. localhost:8433) and writing `GET /stream HTTP/1.0\n\n` will get you the events.

**Reply by hdtodd** [reply]
Am I right that there's no "output HTTP" option for the .conf file? 

I started up the http stream on my "production" rtl_433 Pi and watched the stream from my Mac.  Pretty straightforward.  It looks like I could use either source relatively easily with the rest of the code remaining unchanged.  

I have no idea how to handle disconnects and reconnects.  Another learning opportunity.

Is port 8433 your standard for the HTTP stream?

**Reply by zuckschwerdt** [reply]
In a .conf `output http` should work. Yes, "port 80" plus "rtl_433" gets us `8433` :)
